### PR TITLE
feat: add an intent for importing external events (WPB-10756)

### PR DESCRIPTION
**Pull request by typfel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10756" title="WPB-10756" target="_blank">TaskWPB-10756</a>  [Clients] Build tooling for generating random events for performance testing
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Add an intent for importing external events from a file

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/2971

#### How to Test

Launch the Wire app in debug mode and run the following adb commands:
```
> adb root
> adb push /path/to/events.txt "/data/data/com.wire.android.internal.debug/files"
> adb shell am start -W -a android.intent.action.SYNC -d file:///events.txt
```
----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/WireActivity.kt [hunk 1]
```diff
@@ -525,6 +531,14 @@ class WireActivity : AppCompatActivity() {
         }
     }
 
+    private fun handleSynchronizeExternalData(intent: Intent) {
+        intent.data?.lastPathSegment.let { eventsPath ->
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/ui/WireActivity.kt:554]: Maybe limit this to debug builds and close this door on production?

`if (BuildConfig.DEBUG) {`

Comment (typfel) [reply app/src/main/kotlin/com/wire/android/ui/WireActivity.kt:554]: Done

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -228,6 +228,12 @@
                 <category android:name="android.intent.category.DEFAULT" />
                 <data android:mimeType="application/*" />
             </intent-filter>
+
+            <intent-filter>
+                <action android:name="android.intent.action.SYNC" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:scheme="file" />
+            </intent-filter>
```
Comment (MohamadJaara) [inline app/src/main/AndroidManifest.xml]: you can also move this intent-filter to only debug version manifest 
under app/src/debug create a new `AndroidManifest.xm` and add this to it

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
        xmlns:tools="http://schemas.android.com/tools"
        package="com.wire.android">

    <application android:name=".WireApplication">
        <activity android:name=".ui.WireActivity">
            <intent-filter>
                <action android:name="android.intent.action.SYNC" />
                <category android:name="android.intent.category.DEFAULT" />
                <data android:scheme="file" />
            </intent-filter>
        </activity>
    </application>
</manifest>

```

then when building gradle will merge the main and the debug files together to create the final app manifest

Comment (typfel) [reply app/src/main/AndroidManifest.xml]: Ah cool! Then I can remove the `if (!BuildConfig.DEBUG)` I assume

Comment (MohamadJaara) [reply app/src/main/AndroidManifest.xml]: > Ah cool! Then I can remove the `if (!BuildConfig.DEBUG)` I assume

i would keep it just in case 😅

Comment (typfel) [reply app/src/main/AndroidManifest.xml]: Done

**Review by MohamadJaara — APPROVED** [review]

**Review by vitorhugods — APPROVED** [review]

**Pull request merged**